### PR TITLE
[deckhouse] application conditions summary 

### PR DESCRIPTION
**Pull request by diyliv**
## Description

Replace ad-hoc Application status reporting with a deterministic derivation of
`status.summary` produced **in the same pass that maps the conditions**, from
the internal package pipeline state (RequirementsMet → ReadyOnFilesystem →
Loaded → Configured → HooksProcessed → ManifestsApplied → Scaled).

`summarize` is a pure function that classifies every application into exactly
one canonical lifecycle `state` and attaches a human-readable `message` and an
actionable `tip`:

| State | Meaning |
|---|---|
| `Pending` | First install not started or blocked externally (Installed=False, reason Pending / RequirementsUnmet). |
| `Failed` | No working version serving — first install failed, or an update failed at a late stage. |
| `Updating` | New version installing while the old one still serves. |
| `Ready` | Installed and every primary condition healthy. |
| `Degraded` | Installed, but a reconcile problem surfaced with no active update. |
| `Suspended` | A hard dependency was disabled under a running app. |

The state is **always** present — a healthy app reports `Ready` rather than an
empty summary — so the UI never has to re-implement the state machine on top of
the conditions list.

A new `status.summary` object (state/message/tip) replaces the previous
top-level `status.phase` / `status.message` / `status.howToSolve` fields. Print
columns: `State` by default; `Repository`, `Installed`, `Ready` and `Message`
under `-owide`.

### Design note

`summarize` is fed the **same pre-mapping state the mapper consumed**, not the
merged external conditions. It reuses the mapper's own helpers (`phaseOf`,
`isDependencyDisabled`, the pipeline chains, `canonicalReason`), so there is a
single definition of "what phase are we in", "is a dependency disabled" and
"what failed" — the summary cannot drift from the conditions, and reasons are
read straight from the internal pipeline instead of being reverse-engineered
from the emitted conditions.

## Why do we need it, and what problem does it solve?

Module status (state + message + tip) was derived ad-hoc across different code
paths, leading to inconsistent reporting and unclear operator guidance. This
change provides a single, tested, canonical implementation that:

- Produces the same `state/message/tip` for the same inputs every time.
- Covers every lifecycle scenario (install, update, reconcile, dependency-off).
- Lets the UI and `kubectl` agree on a single `state` value.
- Is testable in isolation (table-driven tests for every scenario + edge cases).
- Surfaces actionable status via `kubectl get applications`.

## Example

`kubectl get app -A` (defaults — `STATE` is always populated):
```
NAMESPACE   NAME          PACKAGE           VERSION   STATE      AGE
default     bug-test      test              v0.0.2    Failed     23h
default     doc-test      test              v0.0.2    Degraded   28h
default     ready-app     cossacks-server   v0.0.1    Ready      30h
default     ready-app-2   cossacks-server   v0.0.1    Updating   3h17m
```

`kubectl get app -A -owide` (adds Repository, Installed, Ready, Message):
```
NAMESPACE   NAME       PACKAGE   VERSION   STATE      REPOSITORY   INSTALLED   READY   MESSAGE                                                                AGE
default     bug-test   test      v0.0.2    Failed     test         False       False   Installation failed: startup or runtime hooks failed                  5m
default     doc-test   test      v0.0.2    Degraded   test         True        True    Reconcile failed: workload health monitor reports degraded            28h
```

Healthy application — `state: Ready`, no message or tip:
```yaml
status:
  conditions:
    - type: Installed             status: "True"
    - type: Ready                 status: "True"
    - type: Scaled                status: "True"
    - type: Managed               status: "True"
    - type: ConfigurationApplied  status: "True"
  summary:
    state: Ready
```

Reconcile degradation — `state: Degraded`:
```yaml
status:
  conditions:
    - type: Installed             status: "True"
    - type: Ready                 status: "True"
    - type: Scaled                status: "False"  reason: Degraded
    - type: Managed               status: "True"
    - type: ConfigurationApplied  status: "True"
  summary:
    state: Degraded
    message: 'Reconcile failed: workload health monitor reports degraded'
    tip: 'Workload health monitor reports degraded. Check pod status and logs to identify the root cause.'
```

Install failure — `state: Failed`:
```yaml
status:
  conditions:
    - type: Installed             status: "False"  reason: HookFailed
    - type: Ready                 status: "False"  reason: HookFailed
    - type: Managed               status: "False"  reason: HookFailed
    - type: ConfigurationApplied  status: "False"  reason: HookFailed
  summary:
    state: Failed
    message: 'Installation failed: startup or runtime hooks failed'
    tip: 'Check the failed hook logs. Fix the configuration or hook code. The attempt will be retried on the next reconcile.'
```

Update failure (old version no longer serving) — `state: Failed`:
```yaml
status:
  currentVersion:
    version: 0.0.1
  conditions:
    - type: Installed             status: "True"
    - type: UpdateInstalled       status: "False"  reason: HookFailed
    - type: Ready                 status: "False"  reason: HookFailed
    - type: Scaled                status: Unknown  reason: HookFailed
    - type: Managed               status: "False"  reason: HookFailed
    - type: ConfigurationApplied  status: "False"  reason: HookFailed
  summary:
    state: Failed
    message: 'Update failed: startup or runtime hooks of the new version failed; previous version is no longer serving'
    tip: 'The application is not serving requests. Check the new version''s hook logs. Fix the hook/config or roll back the module version manually.'
```

Dependency disabled — `state: Suspended`:
```yaml
status:
    conditions:
    - lastTransitionTime: "2026-05-29T20:48:17Z"
      message: 'noneOf group ''testing'' violated: forbidden modules installed: [s-p-test]'
      observedGeneration: 2
      reason: RequirementsUnmet
      status: "False"
      type: Installed
    - lastTransitionTime: "2026-05-29T20:48:17Z"
      message: 'noneOf group ''testing'' violated: forbidden modules installed: [s-p-test]'
      observedGeneration: 2
      reason: RequirementsUnmet
      status: "False"
      type: Ready
    - lastTransitionTime: "2026-05-29T20:48:17Z"
      message: 'noneOf group ''testing'' violated: forbidden modules installed: [s-p-test]'
      observedGeneration: 2
      reason: RequirementsUnmet
      status: Unknown
      type: Managed
    - lastTransitionTime: "2026-05-29T20:48:17Z"
      message: 'noneOf group ''testing'' violated: forbidden modules installed: [s-p-test]'
      observedGeneration: 2
      reason: RequirementsUnmet
      status: Unknown
      type: ConfigurationApplied
    - lastTransitionTime: "2026-05-29T20:48:17Z"
      message: 'noneOf group ''testing'' violated: forbidden modules installed: [s-p-test]'
      observedGeneration: 2
      reason: RequirementsUnmet
      status: Unknown
      type: Scaled
    summary:
      message: 'Application is suspended: requirements unmet'
      state: Suspended
      tip: Solve the application requirements. After it, the controller will automatically restore all conditions and resume operation..
```

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse
type: feature
summary: Added a conditions summary to applications.
```

### CODE REVIEW

**Review by z9r5 — APPROVED** [review]

File: deckhouse-controller/crds/application.yaml [hunk 1]
```diff
@@ -166,6 +167,15 @@ spec:
                   successful apply.
                 type: object
                 x-kubernetes-preserve-unknown-fields: true
+              message:
+                description: Message is a human-readable description of the current status.
+                type: string
+              howToSolve:
```
Comment (z9r5) [inline deckhouse-controller/crds/application.yaml]: There is no translation

**Review by ipaqsa — APPROVED** [review]

**Review by ldmonster — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [20317](https://github.com/deckhouse/deckhouse/pull/20317) to the branch [release-1.76](https://github.com/deckhouse/deckhouse/tree/release-1.76) successful!

**Comment by deckhouse-BOaTswain** [comment]
Backport failed ⚠️.
